### PR TITLE
Reset `webdev`, `dwds`, `test_common` after publishing

### PR DESCRIPTION
**Pull request by elliette**
Also closes https://github.com/dart-lang/webdev/issues/2040 to use `pubspec_overrides` for packages

### CODE REVIEW

File: test_common/pubspec.yaml [hunk 1]
```diff
@@ -6,16 +6,11 @@ environment:
   sdk: ">=3.0.0-134.0.dev <4.0.0"
 
 dependencies:
-  dwds: 18.0.2
+  dwds: ^18.0.0
```
Comment (annagrin) [inline test_common/pubspec.yaml:9]: I wonder if we could just say "any" so this never breaks? This package is never published and the packages that import it have their own dwds constraints.

File: frontend_server_common/pubspec_overrides.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+dependency_overrides:
```
Comment (annagrin) [inline frontend_server_common/pubspec_overrides.yaml:1]: Is this file needed? Looks like we already import local dwds in the pubspec file.

**Review by annagrin — COMMENTED** [review]
Thanks Elliott, this is very helpful for releases! 

I left a couple of comments.

File: frontend_server_client/pubspec.yaml [hunk 3]
```diff
@@ -21,8 +21,3 @@ dev_dependencies:
   test_descriptor: ^2.0.0
   test_process: ^2.0.0
   vm_service: ^8.0.0
-
-dependency_overrides:
```
Comment (annagrin) [inline frontend_server_client/pubspec.yaml:25]: Can this break on release of frontent_server_client if the condition in the comment is not met yet?

Comment (elliette) [reply frontend_server_client/pubspec.yaml:25]: Hmm I think this can now be removed, I see https://github.com/dart-lang/test/pull/1753 and https://github.com/dart-lang/webdev/issues/1762#issuecomment-1285916352 both indicating that. @jakemac53 to confirm

File: frontend_server_common/pubspec_overrides.yaml (hunk 2 above)
Comment (elliette) [reply frontend_server_common/pubspec_overrides.yaml]: Good point, deleted this file!

File: test_common/pubspec.yaml (hunk 1 above)
Comment (elliette) [reply test_common/pubspec.yaml]: Done

File: frontend_server_common/pubspec_overrides.yaml (hunk 2 above)
Comment (annagrin) [reply frontend_server_common/pubspec_overrides.yaml:1]: That's strange:) Some weirdness of the constraint solver I guess. We can also try "any" for dwds in the pubspec (since this package is never published an its users already have their constraints for dwds I think it will work), or bring the pubspec_overrides back, if you you prefer.

**Review by annagrin — APPROVED** [review]
I left another comment but it looks good anyway!

File: frontend_server_common/pubspec_overrides.yaml (hunk 2 above)
Comment (annagrin) [reply frontend_server_common/pubspec_overrides.yaml:1]: Actually I think we just can follow what you did for test_common - change the pubspec one to "any" and keep the pubspec_overrides?
